boards/itsybitsy_m4/examples/sercom_interrupt.rs

```diff
@@ -0,0 +1,159 @@
+#![no_std]
+#![no_main]
+
+//! The idea of this example is to show two different things
+//! - How to manually configure SERCOM for a desired mode (UART in this case)
+//! - How to enable an use SERCOM interrupts
+//! The second one is particularly handy because some drivers will give
+//! you a `poll()` or `update()` function, and you are expected to call it  
+//! on the receive interrupt of the peripheral i.e. SERCOM UART RXC
+//! (receive complete flag).
+//!
+//! This example is basically just a blinky script. The MCU sends itself
+//! a byte via UART that will trigger the RXC interrupt and then toggles
+//! the builtin red led
+//!
+//! Wiring
+//! - RX - A4
+//! - TX - A5
+//!
+//! > In the ATSAMX5x MCUs SERCOM UART configuration requires at least
+//! two pins from the same IOSET. You can find more info about this in
+//! the hal documentation or in the MCU datasheet at:
+//! <https://www.microchip.com/en-us/product/ATSAMD51G19A>
+
+use itsybitsy_m4 as bsp;
+#[cfg(not(feature = "use_semihosting"))]
+use panic_halt as _;
+#[cfg(feature = "use_semihosting")]
+use panic_semihosting as _;
+
+use bsp::entry;
+use cortex_m::{peripheral::NVIC, prelude::_embedded_hal_blocking_delay_DelayMs};
+
+use bsp::hal::{
+    clock::GenericClockController,
+    delay::Delay,
+    ehal::{
+        digital::v2::ToggleableOutputPin,
+        serial::{Read, Write},
+    },
+    gpio::v2::{AlternateD, Pin, PushPullOutput, PA04, PA06, PA22},
+    pac::{self, interrupt, CorePeripherals, Peripherals},
+    sercom::v2::{
+        uart::{self, BaudMode, Flags, Oversampling},
+        IoSet3, Sercom0,
+    },
+    time::Hertz,
+};
+
+/// Pin labeled as A5
+type UartRx0 = Pin<PA06, AlternateD>;
+/// Pin labeled as A4
+type UartTx0 = Pin<PA04, AlternateD>;
+type UartPads0 = uart::Pads<Sercom0, IoSet3, UartRx0, UartTx0>;
+type Uart0 = uart::Uart<uart::Config<UartPads0>, uart::Duplex>;
+
+/// Utility function for setting up SERCOM0 pins as an additional
+/// UART peripheral.
+pub fn uart0(
+    clocks: &mut GenericClockController,
+    baud: impl Into<Hertz>,
+    sercom0: pac::SERCOM0,
+    mclk: &mut pac::MCLK,
+    uart_rx: impl Into<UartRx0>,
+    uart_tx: impl Into<UartTx0>,
+) -> Uart0 {
+    let gclk0 = clocks.gclk0();
+    let clock = &clocks.sercom0_core(&gclk0).unwrap();
+    let baud = baud.into();
+    let pads = uart::Pads::default().rx(uart_rx.into()).tx(uart_tx.into());
+    uart::Config::new(mclk, sercom0, pads, clock.freq())
+        .baud(baud, BaudMode::Fractional(Oversampling::Bits16))
+        .enable()
+}
+
+/// For this example we _know_ nothing is going to break
+/// because of concurrency, but for something more
+/// complex this is clearly unsafe.
+/// If you want to know how to share data safely,
+/// give this a read: <https://docs.rust-embedded.org/book/concurrency/>
+static mut LED: Option<Pin<PA22, PushPullOutput>> = None;
+static mut SERIAL: Option<bsp::Uart> = None;
```

Comment:
I feel like we should just use a [critical section mutex](https://docs.rs/cortex-m/0.7.3/cortex_m/interrupt/struct.Mutex.html) to make this safe. Its easy enough to make this safe so we don't need the long caveat

Reply:
or  a [ref cell](https://doc.rust-lang.org/core/cell/struct.RefCell.html)

Reply:
Agreed, I'd rather we don't encourage unsafe practices in our examples when possible.

Reply:
Perhaps we could change that in the USB examples as well? 
In other PR ofc. Is just changing examples to  use a Mutex for the resource so it shouldn't break stuff, so it could be done for all the boards with the usb_serial example

Reply:
Yes, that could be a nice cleanup effort.

Reply:
I already updated the example to use `Mutex<RefCell<Option<_>>>`
If you think that's fine I can start working on the USB examples update then

Reply:
Yup, I think your solution looks good!